gstd/src/async_runtime/locks.rs

```diff
@@ -173,35 +179,58 @@ impl LocksMap {
 
     /// Lock message.
     pub fn lock(&mut self, message_id: MessageId, waiting_reply_to: MessageId, lock: Lock) {
-        let locks = self.0.entry(message_id).or_insert_with(Default::default);
-        locks.insert(LockContext::ReplyTo(waiting_reply_to), lock);
+        self.message_locks(message_id)
+            .insert(LockContext::ReplyTo(waiting_reply_to), lock);
     }
 
     /// Remove message lock.
     pub fn remove(&mut self, message_id: MessageId, waiting_reply_to: MessageId) {
-        let locks = self.0.entry(message_id).or_insert_with(Default::default);
-        locks.remove(&LockContext::ReplyTo(waiting_reply_to));
+        self.message_locks(message_id)
+            .remove(&LockContext::ReplyTo(waiting_reply_to));
     }
 
     /// Inserts a lock for putting a message into sleep.
-    pub fn insert_sleep(&mut self, message_id: MessageId, until_block: u32) {
-        let locks = self.0.entry(message_id).or_insert_with(Default::default);
+    pub fn insert_sleep(&mut self, message_id: MessageId, wake_up_at: BlockNumber) {
+        let locks = self.message_locks(message_id);
         let current_block = exec::block_height();
-        if current_block < until_block {
+        if current_block < wake_up_at {
             locks.insert(
-                LockContext::Sleep(until_block),
-                Lock::exactly(until_block - current_block)
+                LockContext::Sleep(wake_up_at),
+                Lock::exactly(wake_up_at - current_block)
                     .expect("Never fails with block count > 0"),
             );
         } else {
-            locks.remove(&LockContext::Sleep(until_block));
+            locks.remove(&LockContext::Sleep(wake_up_at));
         }
     }
 
     /// Removes a sleep lock.
-    pub fn remove_sleep(&mut self, message_id: MessageId, until_block: u32) {
-        let locks = self.0.entry(message_id).or_insert_with(Default::default);
-        locks.remove(&LockContext::Sleep(until_block));
+    pub fn remove_sleep(&mut self, message_id: MessageId, wake_up_at: BlockNumber) {
+        self.message_locks(message_id)
+            .remove(&LockContext::Sleep(wake_up_at));
+    }
+
+    pub(crate) fn insert_mx_lock_monitor(
+        &mut self,
+        message_id: MessageId,
+        mutex_id: MutexId,
+        wake_up_at: BlockNumber,
+    ) {
+        let locks = self.message_locks(message_id);
+        locks.insert(
+            LockContext::MxLockMonitor(mutex_id),
```

Comment:
and where is it processed?

Reply:
It is processed same way as all the other locks when their `wait` method is called. The purpose of this lock is waking up a message owning an mx lock after its ownership period elapses.